### PR TITLE
feat(academy): add domain contracts and validation

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

- Add the first mobile Academy domain boundary with typed contracts for articles, sections, content blocks, glossary terms, source references, semantic links, search entries, and future retrieval/chatbot preparation.
- Add pure validation rules for article metadata, sensitive sourced content, review requirements, links, source references, render blocks, section targets, glossary terms, and whole-corpus consistency.
- Add an editorial source adapter for parsed Markdown/front matter objects, mapping snake_case source metadata to the domain contracts while validating source registry hydration and slug/file-name consistency.
- Add pilot editorial source files under `docs/academy/` for `houblons`, `levures`, `eau`, glossary terms, and source references.
- Add a first Markdown/YAML parser and deterministic generated corpus output for the pilot Academy content.
- Add repository/use-case boundaries so future screens can read generated Academy content without importing generated files directly.
- Add a generic Academy article block renderer and progressively use it for published generated article details while legacy topics remain available.
- Add a first Academy hub presenter so published generated articles can replace matching legacy cards while unmigrated topics remain available.
- Add local hub search filtering with a mobile search field, clear action, and empty state.
- Improve the generated article reader presentation with localized metadata badges, visible learning objectives, clearer source cards, and tone-aware callouts.
- Allow published generated articles to render even before a matching legacy hardcoded topic exists.
- Add focused unit tests for valid and invalid article, glossary, corpus, editorial source, parser, generation, repository, search, link-resolution, renderer, hub presentation, generated-detail bridge, generated-only article details, and hub search scenarios.

## Context

This is the first implementation slice after ADR-0023 and the Academy design pack.

It does not remove the current hardcoded `features/tools` Academy detail content yet. The goal is to establish the Clean Architecture domain foundation, source-validation boundary, deterministic generated payload, read-use-case boundary, generic renderer, progressive hub/detail migration, mobile-first article discovery, and the first reader UX polish before deleting legacy content.

The Markdown parser is intentionally narrow for this first slice. It supports the pilot source format only: YAML front matter, `## ... {#section-id}` sections, paragraphs, and explicit Academy directives for glossary references, calculator CTAs, and related articles. Unsupported directives fail with actionable file/line errors.

## Checklist

- [x] Happy path + sad path + edge cases covered (tests when code changes)
  - `academy.validation.test.ts` covers 12 article, glossary, and corpus validation scenarios.
  - `academy-source.validation.test.ts` covers 5 editorial source hydration and validation scenarios.
  - `academy-generation.test.ts` covers 5 parser/generator scenarios, including CRLF front matter, repeated directive attributes, unsupported directives, and validation failures before generation.
  - `academy.use-cases.test.ts` covers 4 repository-backed read, search, and link-resolution scenarios.
  - `academy.repository.test.ts` covers 2 generated repository scenarios.
  - `AcademyArticleRenderer.test.tsx` covers renderer metadata, learning objectives, callouts, sources, and semantic interaction scenarios.
  - `academy-hub.presenter.test.ts` covers 3 generated/legacy hub card merge and filtering scenarios.
  - `AcademyHubScreen.test.tsx` and `AcademyTopicDetailsScreen.test.tsx` keep the current hub/details routing behavior, generated-detail bridge, generated-only article detail, and hub search interactions covered.
  - Total targeted Academy and legacy Academy bridge coverage in this PR: 50 passing tests.
- [x] `npm run ci:all` (or package-local equivalent) passes locally
  - Ran targeted mobile checks locally: `npm -w packages/mobile-app test -- --runInBand src/features/academy src/features/tools/presentation/__tests__/AcademyHubScreen.test.tsx src/features/tools/presentation/__tests__/AcademyTopicDetailsScreen.test.tsx`, `npm -w packages/mobile-app run typecheck`, `npm -w packages/mobile-app run lint`, `npm -w packages/mobile-app run format:check`.
- [x] No `any`, no default exports for screens/use-cases/API modules
  - New Academy domain/source/generation/application/data/presentation files use named exports only and no `any`.
- [x] No inline styles, no hardcoded colors, spacing, or font values in `packages/mobile-app/` (use design tokens)
  - The new renderer and hub/detail bridges use `StyleSheet.create()` and existing theme tokens.
- [x] No new ADR violations (see [docs/architecture/decisions/](docs/architecture/decisions/))
  - Implements ADR-0023 domain/source/generation/read-use-case/renderer/hub/detail foundation without adding DB, CMS, backend publishing, vector search, chatbot runtime, or full legacy deletion.
- [ ] Visual emulator review
  - Pending because this Codex session cannot start/control ADB (`Operation not permitted` in the sandbox). The branch is ready for manual emulator review from a local terminal.
- [ ] `PROJECT_LOG.md` will be updated after merge (entry per the [`project-log-entry`](.claude/skills/project-log-entry/SKILL.md) skill)
  - To be done after this PR is merged.

## Linked issue

No linked issue.

### CODE REVIEW

File: packages/mobile-app/src/features/academy/data/generation/academy-markdown-parser.ts [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+  }
+
+  const attributes: Record<string, string> = {};
+  const rawAttributes = match[2];
+  let attributeMatch = ATTRIBUTE_PATTERN.exec(rawAttributes);
+
+  while (attributeMatch) {
+    attributes[attributeMatch[1]] = attributeMatch[2];
+    attributeMatch = ATTRIBUTE_PATTERN.exec(rawAttributes);
+  }
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/features/academy/data/generation/academy-markdown-parser.ts:218]: Fixed in `2b2a6f9`: `parseDirective` now creates a fresh attribute `RegExp` for each directive parse instead of reusing global `/g` state, and `academy-generation.test.ts` covers repeated directives so attributes are parsed correctly for both `glossaryReference` and `calculatorCta` blocks.

File: packages/mobile-app/src/features/academy/data/generation/academy-markdown-parser.ts [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+  readonly value: TValue | null;
+  readonly errors: readonly string[];
+}
+
+interface MarkdownDirective {
+  readonly type: string;
+  readonly attributes: Readonly<Record<string, string>>;
+}
+
+const FRONT_MATTER_PATTERN = /^---\n([\s\S]*?)\n---\n?/;
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/features/academy/data/generation/academy-markdown-parser.ts]: Fixed in `2b2a6f9`: the front matter delimiter pattern now accepts CRLF through `\r?\n`, and `academy-generation.test.ts` covers CRLF Markdown input.
